src/main/java/com/iexec/core/sms/SmsService.java

```diff
@@ -16,35 +16,59 @@
 
 package com.iexec.core.sms;
 
+import com.iexec.commons.poco.task.TaskDescription;
 import com.iexec.commons.poco.tee.TeeFramework;
 import com.iexec.commons.poco.tee.TeeUtils;
 import com.iexec.commons.poco.utils.BytesUtils;
+import com.iexec.commons.poco.utils.HashUtils;
+import com.iexec.core.chain.IexecHubService;
 import com.iexec.core.chain.SignatureService;
+import com.iexec.core.configuration.ResultRepositoryConfiguration;
 import com.iexec.core.registry.PlatformRegistryConfiguration;
+import com.iexec.core.task.event.TaskInitializedEvent;
 import com.iexec.sms.api.SmsClient;
 import com.iexec.sms.api.SmsClientProvider;
 import feign.FeignException;
 import lombok.extern.slf4j.Slf4j;
 import org.apache.commons.lang3.StringUtils;
+import org.springframework.context.event.EventListener;
 import org.springframework.stereotype.Service;
+import org.web3j.crypto.Hash;
 
 import java.util.Optional;
 
+import static com.iexec.sms.secret.ReservedSecretKeyName.IEXEC_RESULT_IEXEC_RESULT_PROXY_URL;
+
 @Slf4j
 @Service
 public class SmsService {
+    private final IexecHubService iexecHubService;
     private final PlatformRegistryConfiguration registryConfiguration;
+    private final ResultRepositoryConfiguration resultRepositoryConfiguration;
     private final SignatureService signatureService;
     private final SmsClientProvider smsClientProvider;
 
-    public SmsService(PlatformRegistryConfiguration registryConfiguration,
-                      SignatureService signatureService,
-                      SmsClientProvider smsClientProvider) {
+    public SmsService(final IexecHubService iexecHubService,
+                      final PlatformRegistryConfiguration registryConfiguration,
+                      final ResultRepositoryConfiguration resultRepositoryConfiguration,
+                      final SignatureService signatureService,
+                      final SmsClientProvider smsClientProvider) {
+        this.iexecHubService = iexecHubService;
         this.registryConfiguration = registryConfiguration;
+        this.resultRepositoryConfiguration = resultRepositoryConfiguration;
         this.signatureService = signatureService;
         this.smsClientProvider = smsClientProvider;
     }
 
+    private Optional<String> getVerifiedSmsUrl(String chainTaskId) {
```

Comment:
Wouldn't it be better to add final to the chainTaskId parameter for consistency? The same applies to chainTaskId and tag parameters in the getVerifiedSmsUrl function.

Reply:
Thanks, fixed in 174bed6